internal/ir/xds.go

```diff
@@ -309,6 +313,42 @@ type JwtRequestAuthentication struct {
 	Providers []egv1a1.JwtAuthenticationFilterProvider `json:"providers,omitempty" yaml:"providers,omitempty"`
 }
 
+// CorsPolicy holds the Cross-Origin Resource Sharing (CORS) policy for the route.
+//
+// +k8s:deepcopy-gen=true
+type CorsPolicy struct {
+	// AllowOrigins defines the origins that are allowed to make requests.
+	AllowOrigins []*StringMatch `json:"allowOrigins,omitempty" yaml:"allowOrigins,omitempty"`
+	// AllowMethods defines the methods that are allowed to make requests.
+	AllowMethods []string `json:"allowMethods,omitempty" yaml:"allowMethods,omitempty"`
+	// AllowHeaders defines the headers that are allowed to be sent with requests.
+	AllowHeaders []string `json:"allowHeaders,omitempty" yaml:"allowHeaders,omitempty"`
+	// ExposeHeaders defines the headers that can be exposed in the responses.
+	ExposeHeaders []string `json:"exposeHeaders,omitempty" yaml:"exposeHeaders,omitempty"`
+	// MaxAge defines how long the results of a preflight request can be cached.
+	MaxAge *metav1.Duration `json:"maxAge,omitempty" yaml:"maxAge,omitempty"`
+	// AllowCredentials defines whether the resource allows credentials.
+	// Defaults to false.
+	AllowCredentials bool `json:"allowCredentials,omitempty" yaml:"allowCredentials,omitempty"`
+	// AllowPrivateNetwork defines whether allow whose target server’s IP address
+	// is more private than that from which the request initiator was fetched.
+	// Defaults to false.
+	AllowPrivateNetworkAccess bool `json:"allowPrivateNetwork,omitempty" yaml:"allowPrivateNetwork,omitempty"`
+}
+
+func (c *CorsPolicy) Validate() error {
+	var errs error
+
```

Comment:
imo all fields within `cors` are optional and these zero length checks are not required

Reply:
In theory, we can, but it's meaningless if you set a CorsPolicy but leave the domains or methods empty.

Reply:
Rethinking it, it can be useful when the backend service allows CORS and we want to explicitly disable it at the Gateway.  We should make them optional.

Reply:
I agree with you reg domain, we should really consider making it non option in the API